### PR TITLE
No-op operation on multiple register for the same type

### PR DESCRIPTION
**Pull request by spydon**

### CODE REVIEW

File: lib/queryable_ordered_set.dart [hunk 1]
```diff
@@ -65,6 +69,9 @@ class QueryableOrderedSet<T> extends OrderedSet<T> {
     return result.data as List<C>;
   }
 
+  /// Whether type [C] is registered as a cache
+  bool isRegistered<C>() => _cache.containsKey(C);
```
Comment (luanpotter) [inline lib/queryable_ordered_set.dart:73]: can we add a test for this method?

Comment (spydon) [reply lib/queryable_ordered_set.dart:73]: Good idea, added!

File: test/queryable_ordered_set_test.dart [hunk 2]
```diff
@@ -230,6 +230,30 @@ void main() {
...
+        final orderedSet = QueryableOrderedSet<Animal>(
+          Comparing.on((e) => e.name),
+        );
+        // No caches should be registered on a clean set
+        expect(orderedSet.isRegistered<Animal>(), false);
+        expect(orderedSet.isRegistered<Mammal>(), false);
+        orderedSet.register<Animal>();
+        // The Animal cache should report as registered after it has been
+        // registered
+        expect(orderedSet.isRegistered<Animal>(), true);
```
Comment (luanpotter) [inline test/queryable_ordered_set_test.dart:245]: expect(orderedSet.isRegistered<Mammal>(), false);

**Review by luanpotter — APPROVED** [review]
LGTM w one comment

File: test/queryable_ordered_set_test.dart [hunk 3]
```diff
@@ -230,6 +230,30 @@ void main() {
...
+        // registered
+        expect(orderedSet.isRegistered<Mammal>(), true);
+        // The Animal cache should still be reported as registered after another
+        // cache has been registered
+        expect(orderedSet.isRegistered<Animal>(), true);
+        orderedSet.register<Animal>();
+        // Both caches should still be reported as registered after a cache has
+        // been re-registered (no-op)
+        expect(orderedSet.isRegistered<Animal>(), true);
+        expect(orderedSet.isRegistered<Mammal>(), true);
```
Comment (spydon) [inline test/queryable_ordered_set_test.dart:257]:
```suggestion
        expect(orderedSet.isRegistered<Mammal>(), true);
        // A call to isRegistered without a type should always be false
        expect(orderedSet.isRegistered(), false);
```

File: test/queryable_ordered_set_test.dart (hunk 2 above)
Comment (spydon) [reply test/queryable_ordered_set_test.dart:245]: Added it in the end of the test :+1:

Comment (luanpotter) [reply test/queryable_ordered_set_test.dart:245]: but here it 2would test that registering Animal (a super class) doesn't trigger true to Mammal. in the end they are both true bc you register both by then

Comment (spydon) [reply test/queryable_ordered_set_test.dart:245]: Aaah, nooow I get what you wanted me to test. I guess you forgot the `<Mammal>` part in your comment?

Comment (spydon) [reply test/queryable_ordered_set_test.dart:245]: There we go, now it should hopefully be fine!
It is complaining about something weird in the dartdoc step though.

Comment (luanpotter) [reply test/queryable_ordered_set_test.dart:245]: AH markdown! so sorry hahaha

**Pull request merged**